courses/api.py

```diff
@@ -579,3 +581,34 @@ def is_program_text_id(item_text_id):
         bool: True if the given id is a program id
     """
     return item_text_id.startswith(PROGRAM_TEXT_ID_PREFIX)
+
+
+def process_course_run_grade_certificate(course_run_grade):
+    """
+    Ensure that the course run certificate is in line with the values in the course run grade
+
+    Args:
+        course_run_grade (courses.models.CourseRunGrade): The course run grade for which to generate/delete the certificate
+
+    Returns:
+        (courses.models.CourseRunCertificate, bool, bool) that depicts the CourseRunCertificate, created, deleted values
+    """
+    user = course_run_grade.user
+    course_run = course_run_grade.course_run
+
+    # A grade of 0.0 indicates that the certificate should be deleted
+    should_delete = not bool(course_run_grade.grade)
+    should_create = course_run_grade.is_certificate_eligible
+
+    if should_delete:
```

Comment:
I'm not sure that this operation should be deleting certificates and I don't see anything about it in the issue Peter wrote, is there a reason you included this behavior?

Reply:
Most of the things are migrated from xPRO and this has been part of the certificate process there, I also wondered why. And on thing i could think of was that we have an option for resetting grades for a user in edX and in that case if a user gets his grades reset by Course Admins he is going to have `0` grade in which case I suppose we delete the certificate if it has been created already. But we can change this, let me know what you think?

Reply:
That makes sense.